tests/scryer/proptests.rs

```diff
@@ -0,0 +1,85 @@
+use std::num::NonZeroI128;
+
+use dashu::Rational;
+use proptest::prelude::*;
+
+use scryer_prolog::machine::parsed_results::{QueryMatch, QueryResolution, Value};
+
+fn generate_query_resolution() -> impl Strategy<Value = QueryResolution> {
+    prop_oneof![
+        Just(QueryResolution::False),
+        Just(QueryResolution::True),
+        prop::collection::vec(generate_query_match(), 1..5).prop_map(QueryResolution::Matches)
+    ]
+}
+
+fn generate_query_match() -> impl Strategy<Value = QueryMatch> {
+    prop::collection::btree_map(any::<String>(), generate_value(), 1..5)
+        .prop_map(|map| QueryMatch { bindings: map })
+}
+
+fn float_roundtrips(&orig: &'_ f64) -> bool {
```

Comment:
We might want to enable the `float_roundtrip` feature for the `serde_json` crate to ensure all f64 roundtrip perfectly